src/Core/Project.fs

```diff
@@ -140,16 +160,121 @@ module Project =
             | null -> []
             | rootPath -> findProjs rootPath
 
-
-
         cached |> Seq.iter fs.unlinkSync
         window.showInformationMessage("Cache cleared")
 
+    let pickFSACWorkspace (ws: WorkspacePeekFound array) =
+        let m (x: WorkspacePeekFound) =
+            match x with
+            | WorkspacePeekFound.Directory dir ->
+                (sprintf "[DIR] %s" dir.Directory), x
+            | WorkspacePeekFound.Solution sln ->
+                let relativeSln = path.relative (workspace.rootPath, sln.Path)
+                (sprintf "[SLN] %s" relativeSln), x
+        let projects = ws |> Array.map m |> Map.ofArray
+        if projects |> Map.isEmpty then
+            None |> Promise.lift
+        else
+            promise {
+                let opts = createEmpty<QuickPickOptions>
+                opts.placeHolder <- Some "Place"
+                let chooseFrom = projects |> Map.toArray |> Array.map fst |> ResizeArray
+                let! chosen = window.showQuickPick(chooseFrom |> Case1, opts)
+                return if JS.isDefined chosen
+                       then projects |> Map.tryFind chosen 
+                       else None
+            }
+
+    let rec foldFsproj (item: WorkspacePeekFoundSolutionItem) =
+        match item.Kind with
+        | WorkspacePeekFoundSolutionItemKind.Folder folder ->
+            folder.Items |> Array.collect foldFsproj
+        | WorkspacePeekFoundSolutionItemKind.MsbuildFormat msbuild ->
+            [| item.Name, msbuild |]
+
+    let private getWorkspaceForModeDirectory () =
+        promise {
+            let fsprojs = findAll ()
+            let wdir =
+                { WorkspacePeekFoundDirectory.Directory = vscode.workspace.rootPath
+                  Fsprojs = fsprojs |> Array.ofList }
+            return WorkspacePeekFound.Directory wdir
+        }
+    
+    let private getWorkspaceForMode mode =
+        let peek () = LanguageService.workspacePeek (vscode.workspace.rootPath) 2 (excluded |> List.ofArray)
```

Comment:
Choice seems fine. But maybe it would be worth using DU / Enum instead so we know what means the value we're passing here.